### PR TITLE
Simplify by removing intermediate FIAT coordinates

### PR DESCRIPTION
**Pull request by chrisrichardson**
* Simplifies some code generation by not creating intermediate FIAT coordinates in code

### CODE REVIEW

File: ffcx/codegeneration/evaluatebasis.py [hunk 1]
```diff
@@ -413,14 +410,14 @@ def _idx2d(p, q):
...
-        code += [L.VariableDecl("const double", f2, (1.0 - Y[1]) / 2.0)]
+        code += [L.VariableDecl("const double", f2, (1.0 - Y1) / 2.0)]
         code += [L.VariableDecl("const double", f3, f2 * f2)]
     for r in range(1, embedded_degree):
         rr = _idx2d((r + 1), 0)
         ss = _idx2d(r, 0)
         tt = _idx2d((r - 1), 0)
-        A = (2 * r + 1.0) / (r + 1)
-        B = r / (1.0 + r)
+        A = 1 + r / (r + 1)
```
Comment (michalhabera) [inline ffcx/codegeneration/evaluatebasis.py:419]: It this correct? Shouldn't it be `2 - 1/(r+1)`?

Comment (chrisrichardson) [reply ffcx/codegeneration/evaluatebasis.py:419]: I think it is correct, `(2*r +1)/(r+1) = (r + (r+1))/(r+1) = 1 + r/(r+1)`. But maybe I shouldn't touch it. So many of these expressions are rather cryptic.

Comment (michalhabera) [reply ffcx/codegeneration/evaluatebasis.py:419]: Ah, ok, both expressions are correct. Yes, many of these formulas are hard to follow without some additional material...
